File: frontend/src/types/cluster.ts
export type ClusterOverview = {
  cluster_id: string;
  nodes: { total: number; running: number; pending: number; failed: number; succeeded: number; unknown: number };
  namespaces: number;
  pods: { total: number; running: number; pending: number; failed: number; succeeded: number; unknown: number };
  deployments: number;
  services: number;
  warning_events: number;
  gpu: { gpu_nodes: number; total_gpu: number; requested_gpu: number; gpu_pods: number };
};

File: backend/app/schemas/cluster.py
from pydantic import BaseModel


class StatusCount(BaseModel):
    total: int
    running: int = 0
    pending: int = 0
    failed: int = 0
    succeeded: int = 0
    unknown: int = 0


class GpuSummary(BaseModel):
    gpu_nodes: int
    total_gpu: int
    requested_gpu: int
    gpu_pods: int


class ClusterOverview(BaseModel):
    cluster_id: str
    nodes: StatusCount
    namespaces: int
    pods: StatusCount
    deployments: int
    services: int
    warning_events: int
    gpu: GpuSummary
